source/type/pb_converter.cpp

```diff
@@ -49,6 +49,636 @@ namespace nt = navitia::type;
+}
+
+
+void PbCreator::Filler::fill_pb_object(const navitia::type::MultiLineString* shape,
+                                         pbnavitia::MultiLineString* geojson){
+    for (const std::vector<nt::GeographicalCoord>& line: *shape) {
+        auto l = geojson->add_lines();
+        for (const auto coord: line) {
+            auto c = l->add_coordinates();
+            c->set_lon(coord.lon());
+            c->set_lat(coord.lat());
+        }
+    }
+}
+
+
+void PbCreator::Filler::fill_pb_object(const navitia::type::GeographicalCoord* coord,
+                                         pbnavitia::Address* address){
+    if (!coord->is_initialized()) {
+        return;
+    }
+
+    try{
+        const auto nb_way = pb_creator.data.geo_ref->nearest_addr(*coord);
+        house_number_coord hn = {nb_way.first, *coord};
+        const way_pair waypair = {nb_way.second, hn};
+        fill(&waypair, address);
+    }catch(navitia::proximitylist::NotFound){
+        LOG4CPLUS_DEBUG(log4cplus::Logger::getInstance("logger"),
+                       "unable to find a way from coord ["<< coord->lon() << "-" << coord->lat() << "]");
+    }
+}
+
+void PbCreator::Filler::fill_pb_object(const way_pair* waypair, pbnavitia::Address* address){
+    const navitia::georef::Way* way = waypair->first;
+    const way_pair_name waypair_name = {{way, way->name}, waypair->second};
+    fill(&waypair_name, address);
+}
+
+void PbCreator::Filler::fill_pb_object(const way_pair_name* waypair_name, pbnavitia::Address* address){
+    const way_name wayname = waypair_name->first;
+    const house_number_coord hn_coord = waypair_name->second;
+
+    if(wayname.first == nullptr)
+        return ;
+    int new_depth = (depth <= 3) ? depth : 3;
+    address->set_name(wayname.second);
+    std::string label;
+    if(hn_coord.first >= 1){
+        address->set_house_number(hn_coord.first);
+        label += std::to_string(hn_coord.first) + " ";
+    }
+//    label += get_label(wayname.first);
+//    address->set_label(label);
+
+    if(hn_coord.second.is_initialized()) {
+        address->mutable_coord()->set_lon(hn_coord.second.lon());
+        address->mutable_coord()->set_lat(hn_coord.second.lat());
+        std::stringstream ss;
+        ss << std::setprecision(16) << hn_coord.second.lon() << ";" << hn_coord.second.lat();
+        address->set_uri(ss.str());
+    }
+
+    if(new_depth > 0){
+        fill(wayname.first->admin_list, [&](){return address->add_administrative_regions();});
+    }
+}
+
+void PbCreator::Filler::fill_pb_object(const nt::StopPointConnection* c, pbnavitia::Connection* connection){
+    if(c == nullptr)
+        return ;
+    int new_depth = (depth <= 3) ? depth : 3;
+
+    connection->set_duration(c->duration);
+    connection->set_display_duration(c->display_duration);
+    connection->set_max_duration(c->max_duration);
+    if(c->departure != nullptr && c->destination != nullptr && new_depth > 0){
```

Comment:
the nullptr tests should not be done here. They should be done in the fill method.